internal/graph/checkingresourcestream.go

```diff
@@ -511,21 +519,51 @@ func (crs *checkingResourceStream) spawnIfAvailable() {
 
 // queue queues a reachable resources result to be processed by one of the processing worker(s), before publishing.
 func (crs *checkingResourceStream) queue(result *v1.DispatchReachableResourcesResponse) bool {
-	crs.rq.addPossibleResource(possibleResource{
+	currentResource := possibleResource{
 		reachableResult: result,
 		lookupResult:    nil,
 		orderingIndex:   crs.reachableResourcesCount,
-	})
+	}
+
+	if result.Resource.ResultStatus == v1.ReachableResource_HAS_PERMISSION {
```

Comment:
my understanding is this is an optimization to make sure we do not enqueue work if a reachable resource was determined to have permission. Can we update the method documentation accordingly? `addPossibleResource` method documentation may need adjustment too

Reply:
Correct. I'll update the documentation